piet-cairo/src/text.rs

```diff
@@ -129,9 +130,20 @@ impl TextLayout for CairoTextLayout {
         self.width
     }
 
-    fn update_width(&mut self, new_width: f64) -> Result<(), Error> {
-        self.width = new_width;
+    // TODO refactor this to use same code as new_text_layout
+    fn update_width(&mut self, new_width: impl Into<Option<f64>>) -> Result<(), Error> {
+        let new_width = new_width.into().unwrap_or(std::f64::INFINITY);
+
         self.line_metrics = lines::calculate_line_metrics(&self.text, &self.font, new_width);
+
+        let widths = self.line_metrics.iter().map(|lm| {
```

Comment:
is this fixing a separate bug?

Reply:
Yes, I made a quick note but I can say a bit more here. In the text layout build step, I calculate the overall layout width by using the max of the widths of the lines in the layout. Unfortunately, on the first pass, I forgot that this also needed to be done when updating the width (I had a placeholder which simply set the width to the `width` argument).

I thought about making it cleaner by doing some refactoring; but decided to keep it simple for now, and perhaps refactor in another PR.

Reply:
I also thought about fixing in a separate PR, I can also do that if you prefer. I guess I just wanted to fix it as soon as I saw it :sweat_smile:

Reply:
I understand, separate PR is probably strictly 'better' but i'm happy to get this in, just wanted to make sure I wasn't overlooking something.